server/src/utils/operatingSystem.ts

```diff
@@ -10,13 +10,66 @@ export async function runCmd(cmd: string, cwd?: string): Promise<string> {
     exec(cmd, { cwd }, function (error, stdout) {
       if (error !== null) {
         reject(error);
+        return;
       }
 
       resolve(stdout);
     });
   });
 }
 
+/**
+ * Executes a command using exec, writes provided input to stdin,
+ * and returns a Promise that resolves with stdout and stderr.
+ *
+ * @param {string} command - The command to execute.
+ * @param {string} input - The input string to write to the child process's stdin.
+ * @param {ExecOptions} options - The options to pass to the exec function.
+ * @param {number} timeout - The timeout in milliseconds.
+ * @returns {Promise<{stdout: string, stderr: string}>}
+ */
+export async function execWithInput(
+  command: string,
+  input: string,
+  options: ExecOptions = {}
+): Promise<{ stdout: string; stderr: string }> {
+  return new Promise((resolve, reject) => {
+    const child = exec(command, options, (error, stdout, stderr) => {
+      // `error` is any execution error. e.g. command not found, non-zero exit code, etc.
+      if (error) {
+        reject(error);
+      } else {
+        resolve({ stdout, stderr });
+      }
+    });
+
+    // This could be triggered if node fails to spawn the child process
+    child.on("error", (err) => {
+      reject(err);
+    });
+
+    const stdin = child.stdin;
+
+    if (stdin) {
+      stdin.on("error", (err) => {
+        // This captures EPIPE error
+        reject(err);
+      });
+
+      child.once("spawn", () => {
+        stdin.write(input, (error) => {
+          if (error) {
+            reject(error);
+          }
+          stdin.end();
+        });
+      });
```

Comment:
```suggestion
      child.once("spawn", () => {
        if (!stdin.writable || child.killed) {
          return reject(new Error("Failed to write to unwritable stdin"));
        }

        stdin.write(input, (error) => {
          if (error) {
            reject(error);
          }
          stdin.end();
        });
      });
```

Do we still need a writable/killed guard, or do we think this is covered by the initial spawn step?

Reply:
I had that guard in place but removed it because I didn't find any way to trigger it. But I guess we can just leave it to be extra sure.

Reply:
I would have no problem removing it if we know of no way it could be hit.